m4opt/_cli/schedule.py

```diff
@@ -97,6 +99,22 @@ def schedule(
     ] = 0,
 ):
     """Schedule a target of opportunity observation."""
+
+    mission_mapping = {
+        name: getattr(missions, name)
+        for name in missions.__all__
+        if isinstance(getattr(missions, name), missions.Mission)
+    }
+    # Fetch the mission based on user input
+    selected_mission = mission_mapping.get(mission.lower())
+    if not selected_mission:
+        typer.echo(
+            f"Mission '{mission}' is not recognized. Available options: {', '.join(mission_mapping.keys())}"
+        )
+        raise typer.Exit()
+
+    mission = selected_mission
```

Comment:
What was wrong with the original implementation of the mission option?

Reply:
Since I used the 'mission' option above, which only contains the mission name as a string, I am now retrieving the mission module. I use the raise statement to prevent users from mistyping the mission name or selecting a mission that doesn't exist.